qiskit/pulse/macros.py

```diff
@@ -77,18 +113,67 @@ def measure(
                 "argument. For assistance, the instructions which are defined are: "
                 "{}".format(measure_name, inst_map.instructions)
             ) from ex
-        for time, inst in default_sched.instructions:
-            if inst.channel.index not in qubits:
-                continue
-            if qubit_mem_slots and isinstance(inst, instructions.Acquire):
-                if inst.channel.index in qubit_mem_slots:
-                    mem_slot = channels.MemorySlot(qubit_mem_slots[inst.channel.index])
-                else:
-                    mem_slot = channels.MemorySlot(unused_mem_slots.pop())
-                inst = instructions.Acquire(inst.duration, inst.channel, mem_slot=mem_slot)
-            # Measurement pulses should only be added if its qubit was measured by the user
-            schedule = schedule.insert(time, inst)
+        schedule = combine_schedules(qubits, schedule, default_sched, qubit_mem_slots, measure_group_qubits)
+    return schedule
+
+
+def measure_v2(
+    qubits: List[int],
+    backend=None,
+    target: Optional[Target] = None,
```

Comment:
You can reduce `backend` and `target` by formatting the arguments in `measure` function before dispatching the logic. Please keep interface minimum for maintainability. Same for v1.